microsoft/testsuites/stress/stress_ng_suite.py

```diff
@@ -0,0 +1,171 @@
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "memory")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'vm' class stressors for 60s each.",
+        priority=4,
+    )
+    def verify_stress_ng_vm_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "vm")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'io' class stressors for 60s each.",
+        priority=4,
+    )
+    def verify_stress_ng_io_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "io")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'network' class stressors for 60s each.",
+        priority=4,
+    )
+    def verify_stress_ng_network_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "network")
+
+    def _run_stressor_class(self, environment: Environment, class_name: str) -> None:
+        nodes = [cast(RemoteNode, node) for node in environment.nodes.list()]
+        procs: List[Process] = []
+        try:
+            for node in nodes:
+                procs.append(node.tools[StressNg].launch_class_async(class_name))
+            for proc in procs:
+                proc.wait_result(timeout=self.TIME_OUT, expected_exit_code=0)
+        except Exception as e:
+            self._check_panic(nodes)
+            raise e
+
+    def _run_stress_ng_job(
+        self,
+        job_file: str,
+        environment: Environment,
+        test_result: TestResult,
+    ) -> None:
+        nodes = [cast(RemoteNode, node) for node in environment.nodes.list()]
+        procs: List[Process] = []
+        job_file_name = Path(job_file).name
+        test_status = TestStatus.QUEUED
+        test_msg = ""
+        try:
+            for node in nodes:
+                remote_working_dir = node.working_path / "stress_ng_jobs"
+                node.shell.mkdir(remote_working_dir, exist_ok=True)
+                job_file_dest = remote_working_dir / job_file_name
+                node.shell.copy(PurePath(job_file), job_file_dest)
+                procs.append(node.tools[StressNg].launch_job_async(str(job_file_dest)))
+            for proc in procs:
+                proc.wait_result(expected_exit_code=0)
+            test_status = TestStatus.PASSED
+        except Exception as e:
+            test_status = TestStatus.FAILED
+            test_msg = repr(e)
+        finally:
+            self._send_subtest_msg(
+                test_result.id_,
+                environment,
+                job_file_name,
+                test_status,
+                test_msg,
+            )
+            self._check_panic(nodes)
+
+    def _check_panic(self, nodes: List[RemoteNode]) -> None:
+        for node in nodes:
+            node.features[SerialConsole].check_panic(saved_path=None, force_run=True)
```

Comment:
what will happen if save_path is None?

Reply:
If saved_path is not provided, it will save in a default location (`self._node.local_log_path / get_datetime_path()`). This handling is done in `get_console_log()` of serial_console.py.